pkg/cluster/streams.go

```diff
@@ -184,8 +191,10 @@ func (c *Cluster) generateFabricEventStream(appId string) *zalandov1.FabricEvent
 			Kind:       constants.EventStreamCRDKind,
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        fmt.Sprintf("%s-%s", c.Name, appId),
+			// max length for cluster name is 53 so we can only add 10 more characters / numbers
+			Name:        fmt.Sprintf("%s-%s-%s", c.Name, constants.EventStreamSourceSlotPrefix, util.RandomPassword(5)),
 			Namespace:   c.Namespace,
+			Labels:      c.labelsSet(true),
```

Comment:
use the same labels set what we have for other resources incl. cluster labels like `application: spilo`. Might be debatable but this way it's easier to find stream CRDs created by the Postgres Operator. Alternative: filter by ownerReference, if this is even possible?